go-controller/pkg/cni/helper_linux.go

```diff
@@ -252,60 +252,76 @@ func setupInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInter
 	return hostIface, contIface, nil
 }
 
+// generate a unique interface name for the temporary netdev that will be moved to pod namespace
+// the generated ifname suffix will be "_<ifName>"
+func generateIfName(containerID string, ifName string) string {
```

Comment:
what if the length of the old ifname string is already 15?

Reply:
ifName is the target name for the netdev when moved into the pod (e.g. net1, etc). I don't think it'll ever be 15 chars though. Shall I strip the "net" and use just the number?

Reply:
This is how the ifName is generated after the last commit:
```
func TestGenerateIfName(t *testing.T) {
	containerID := "c123456789012345678901234567890"
	ifname := generateIfName(containerID, "eth0")
	assert.Equal(t, "c123456789012_0", ifname)
	ifname = generateIfName(containerID, "net1")
	assert.Equal(t, "c123456789012_1", ifname)
	ifname = generateIfName(containerID, "net2")
	assert.Equal(t, "c123456789012_2", ifname)
}
```

Reply:
> ifName is the target name for the netdev when moved into the pod (e.g. net1, etc). I don't think it'll ever be 15 chars though. Shall I strip the "net" and use just the number?

it can be specified as any valid linux interface name in the Network Attachment Selection Annotation, see https://docs.google.com/document/d/1Ny03h6IDVy_e_vmElOqR7UdTPAG_RNydhVE1Kx54kFQ/edit section 4.1.2.1.5.

Reply:
I am now generating the interface name in this way: 
`// generate a unique interface name for the temporary netdev that will be moved to pod namespace
func generateIfName(containerID string) string {
	randomId := generateId(5) // random ID with 5 chars
	// ifname max length is 15
	return containerID[:(15-len(randomId))] + randomId
}`
So the resulting temporary ifname will be something like: <leading chars from container-id><random chars>
The length of the ifname will be 15 and it should be unique in the host. Please note this is a temporary name, the interface will be renamed again as soon as it is moved in the pod namespace